components/camel-ai/camel-langchain/src/main/java/org/apache/camel/component/langchain/service/Langchain4jChatHandler.java

```diff
@@ -0,0 +1,132 @@
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ChatMessageType;
+import dev.langchain4j.data.message.SystemMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.input.Prompt;
+import dev.langchain4j.model.input.PromptTemplate;
+import dev.langchain4j.model.output.Response;
+
+public class Langchain4jChatHandler {
+
+    ChatLanguageModel chatLanguageModel;
+
+    public Langchain4jChatHandler(ChatLanguageModel chatLanguageModel) {
+        this.chatLanguageModel = chatLanguageModel;
+    }
+
+    /**
+     * Send one simple message
+     *
+     * @param  message
+     * @return
+     */
+    public String sendMessage(String message) {
+        return this.chatLanguageModel.generate(message);
+    }
+
+    public String sendWithPromptTemplate(String promptTemplate, Map<String, Object> variables) {
+        PromptTemplate template = PromptTemplate.from(promptTemplate);
+        Prompt prompt = template.apply(variables);
+        return this.sendMessage(prompt.text());
+    }
+
+    public String sendWithPromptTemplate(
+            String promptTemplate, Map<String, Object> variables, ChatMessageType chatMessageType) {
+        PromptTemplate template = PromptTemplate.from(promptTemplate);
+        Prompt prompt = template.apply(variables);
+        return this.sendMessage(prompt.text(), chatMessageType);
+    }
+
+    /**
+     * Send a simple message, for a specific Chat Message Type
+     *
+     * @param  message
+     * @param  type
+     * @return
+     */
+    public String sendMessage(String message, ChatMessageType type) {
+        ChatMessage[] chatMessages = { convertChatMessages(type, message) };
+        return this.sendMessage(chatMessages);
+    }
+
+    /**
+     * Send a List of Messages
+     *
+     * @param  type
+     * @param  messages
+     * @return
+     */
+    public String sendMessages(ChatMessageType type, List<String> messages) {
+
+        var chatMessages = messages.stream()
+                .map(message -> (ChatMessage) convertChatMessages(type, message))
+                .toArray(ChatMessage[]::new);
+
+        return sendMessage(chatMessages);
+    }
+
+    /**
+     * Create the messages in the format of Langchain4j ChatMessage
+     *
+     * @param  type
+     * @param  message
+     * @return
+     */
+    private ChatMessage convertChatMessages(ChatMessageType type, String message) {
+        return switch (type) {
+            case USER -> new UserMessage(message);
+            case AI -> new AiMessage(message);
+            case SYSTEM -> new SystemMessage(message);
+            case TOOL_EXECUTION_RESULT -> throw new IllegalArgumentException("Tools are currently not supported");
+        };
+
+    }
+
+    /**
+     * Send And Array of ChatMessages
+     *
+     * @param  messages
+     * @return
+     */
+    public String sendMessage(ChatMessage... messages) {
```

Comment:
not sure if this intended but since this is a varrg, the method can be invoked without parameters

Reply:
I removed this one.